docs/HelloWorldWalkthrough.md

```diff
@@ -0,0 +1,312 @@
+# Hello World VBS Enclave SDK Walkthrough
+
+## Prerequisites:
+ 
+* Windows SDK 10.0.26100.2454 or higher is installed. Latest SDK update is published to:
+  * [https://developer.microsoft.com/en-us/windows/downloads/windows-sdk/](https://developer.microsoft.com/en-us/windows/downloads/windows-sdk/) 
+ 
+* Follow instructions to build the repo to generate the 2 nuget packages:
+  * https://github.com/microsoft/VbsEnclaveTooling?tab=readme-ov-file#build-instructions
+ 
+### Prepare your dev environment
+Full instructions can be found at [VBS Enclaves dev guide](https://learn.microsoft.com/en-us/windows/win32/trusted-execution/vbs-enclaves-dev-guide)
+
+!IMPORTANT! The following steps turn off security features in Windows and allows unsigned drivers to load. Consider only doing so when needed for testing, or setup a test vm. 
+* The device must have a TPM that is enabled in the BIOS.
+* Ensure you are running on a device with Windows 11, version 10.0.26100.2454 or higher
+* Turn off [Secure Boot](https://learn.microsoft.com/en-us/windows-hardware/manufacture/desktop/disabling-secure-boot?view=windows-11)
+  * If Bitlocker is enabled, you will need access to your recovery keys
+* Enable [Test Signing](https://learn.microsoft.com/en-us/windows-hardware/drivers/install/the-testsigning-boot-configuration-option). From an elevated command prompt, run the following and reboot: 
+  * ```BCDEdit /set TESTSIGNING ON```
+* Enable [Memory Integrity](https://learn.microsoft.com/en-us/windows/security/hardware-security/enable-virtualization-based-protection-of-code-integrity?tabs=security). In the Windows Security app, go to Device Security -&gt; Core Integrity -&gt; and toggle Memory Integrity ON, then Reboot.
+
+ 
+* Create a test certificate to use for signing the VBS Enclave. Replace "TheDefaultTestEnclaveCert" with the name you wish to create. eg:
+    ``` 
+    New-SelfSignedCertificate -CertStoreLocation Cert:\\CurrentUser\\My -DnsName "TheDefaultTestEnclaveCert" -KeyUsage DigitalSignature -KeySpec Signature -KeyLength 2048 -KeyAlgorithm RSA -HashAlgorithm SHA256 -TextExtension "2.5.29.37={text}1.3.6.1.5.5.7.3.3,1.3.6.1.4.1.311.76.57.1.15,1.3.6.1.4.1.311.97.814040577.346743379.4783502.105532346
+    ```
+
+## Solution structure
+This sample will create a Host app, a DLL project and include a solution level file as follows:
+- MyHostApp.sln
+  - MySecretEnclave.edl
+  - MyHostApp.vcxproj
+    - main.cpp
+  - MySecretEnclave.vcxproj
+    - MySecretEnclaveExports.cpp
+
+## Build the sample
+### Create the initial host app project and solution
+  1. Start VS, create new C++ Console App named **MyHostApp.vcxproj**
+### Add the Enclave Definition Language file to the solution
+  2. Add the Enclave Definition Language (EDL) file to the solution where you define your interface.
```

Comment:
>2. [](http://example.com/codeflow?start=2&length=2)

use 1. everywhere' for numbered lists

see: [https://github.com/microsoft/VbsEnclaveTooling/blob/main/README.md?plain=1](https://github.com/microsoft/VbsEnclaveTooling/blob/main/README.md?plain=1)